controlplane/kubeadm/internal/cluster.go

```diff
@@ -180,16 +181,15 @@ func (m *Management) getApiServerEtcdClientCert(ctx context.Context, clusterKey
 	return tls.X509KeyPair(crtData, keyData)
 }
 
-type healthCheck func(context.Context) (HealthCheckResult, error)
-
 // HealthCheck will run a generic health check function and report any errors discovered.
 // In addition to the health check, it also ensures there is a 1;1 match between nodes and machines.
-func (m *Management) healthCheck(ctx context.Context, check healthCheck, clusterKey client.ObjectKey) error {
+// To have access to the owned control-plane machines during health checks, need to pass owningMachines here.
+func (m *Management) healthCheck(controlPlane *ControlPlane, nodeChecks HealthCheckResult, clusterKey client.ObjectKey) error {
```

Comment:
I'm finding hard to follow the goal of this func, given that the description states `HealthCheck will run a generic health check function and report any errors discovered` but now the func takes in input `HealthCheckResult`, and it does not run checks anymore

Reply:
Changed description and moved healthcheck to `controller.go`